File: app/src/main/java/com/webianks/hatkemessenger/SMS.java
package com.webianks.hatkemessenger;

/**
 * Created by R Ankit on 24-12-2016.
 */

public class SMS {

    private long _id;
    private String _address;
    private String _msg;
    private String _readState; //"0" for have not read sms and "1" for have read sms
    private long _time;
    private String _folderName;
    private int color;

    public long getId() {
        return _id;
    }

    public String getAddress() {
        return _address;
    }

    public String getMsg() {
        return _msg;
    }

    public String getReadState() {
        return _readState;
    }

    public long getTime() {
        return _time;
    }

    public String getFolderName() {
        return _folderName;
    }


    public void setId(long id) {
        _id = id;
    }

    public void setAddress(String address) {
        _address = address;
    }

    public void setMsg(String msg) {
        _msg = msg;
    }

    public void setReadState(String readState) {
        _readState = readState;
    }

    public void setTime(long time) {
        _time = time;
    }

    public void setFolderName(String folderName) {
        _folderName = folderName;
    }

    @Override
    public boolean equals(Object obj) {

        SMS sms = (SMS) obj;

        return _address.equals(sms._address);
    }

    public int hashCode() {
        return this._address.hashCode();
    }

    public void setColor(int color) {
        this.color = color;
    }

    public int getColor() {
        return color;
    }
}

File: app/src/main/java/com/webianks/hatkemessenger/SMS.kt
package com.webianks.hatkemessenger

/**
 * Created by R Ankit on 24-12-2016.
 */
class SMS {
    var id: Long = 0
    var address: String? = null
    var msg: String? = null
    var readState //"0" for have not read sms and "1" for have read sms
            : String? = null
    var time: Long = 0
    var folderName: String? = null
    var color = 0

    override fun equals(obj: Any?): Boolean {
        val sms = obj as SMS?
        return address == sms!!.address
    }

    override fun hashCode(): Int {
        return address.hashCode()
    }

}